### DISCUSSION TITLE
Option to prefix nzb files in a zipped archive

### DISCUSSION THREAD

**Discussion by EdwinKM** [sabnzbd/sabnzbd#2481] [Ideas]
Redirected from: https://forums.sabnzbd.org/viewtopic.php?p=129165#p129165

I regularly use https://nzbking.com/. With this site you can make zip-packs of multiple (bulk) NZB's.
So, you usually will store this as the name of the series.

nzbking will name all the nzb's in the zip with a vague name (probably a unique database identifier). For example, we store the file as "MySeries.zip", containing:
* asdtadadasd12adssad.nzb
* hgfhewrasdsadas2w4.nzb

Can we create a option to prefix the nzb files with the zip filename?
Like:
* MySeries_asdtadadasd12adssad.nzb
* MySeries_hgfhewrasdsadas2w4.nzb

This is maybe somewhat of a niche request. And i suggest to disable this option by default. However i suspect it should not clutter the code much.

**Comment by thezoggy** [comment]
raw search siites generally are not going to provide you sound nzb collections, and for your usage you prob could just fix with a pre-queue script to extract the nzb/name it how you want before having sab run with it

**Reply by EdwinKM** [reply]
Indeed, but you can create "bulk" collections by selecting the correct checkbox field with nzbking. But mostly still contains false positives. The advantage of this prefix is that you have a clue what to expect. Certainly when you import multiple zip files for multiple series.

However your pre-queue script seems promising. I will check if i have all the properties.

**Reply by EdwinKM** [reply]
I toyed a bit with a pre queue script. And i was afraid of it, but this is at the nzb-level. So, the unpack is already done. As is also stated in the documentation: "You can choose to let SABnzbd run a script just before an NZB enters the queue.".

So, this can only be done at the software level.
Tomorrow i will try to figure out the code, this seems the starting point:
https://github.com/sabnzbd/sabnzbd/blob/develop/sabnzbd/nzbparser.py#L145

**Reply by thezoggy** [reply]
No, i'm saying you handle your niche workflow on your side. Toss your nzb into a folder, your script does the needful then moves it to a watch folder for sab to load / script adds the nzb to sab itself

**Comment by jcfp** [comment]
This specific goal seems rather niche indeed. That said, it could be done from a pre-queue script if the filename of the archive-full-of-nzbs were recorded in the nzo and made available to the script (e.g. as an additional environment var).

**Reply by EdwinKM** [reply]
not sure what the nzo is? Afaik the zip name (something i as user determines) is not registered elsewhere. 

I as toying with the source (on linux in a container) but my added loglines are not logged. I guess it does not read the .py files in runtime.

**Reply by Safihre** [reply]
The nzo is the internal name of an NZB object.
But as far as I know the original filename of the archive is lost indeed.

**Reply by EdwinKM** [reply]
This seems to do the job.
The question will it be accepted upstream? Because than i need to figure out how to add it to the settings screen.

Off course, with a boolean check surrounded. Default not changing the internal nzb name.

![image](https://user-images.githubusercontent.com/1620919/222557084-4eb50802-b7d6-4fd2-86ce-10ccce48f364.png)

**Reply by puzzledsab** [reply]
Here's an example of how to add a switch to the Special page: https://github.com/sabnzbd/sabnzbd/commit/b8d7ec0723e5c9023482d5c4f1b9aaa96fc620a0
It is only the cfg.py and interface.py changes that are needed to add it. The other ones are using it.

**Reply by EdwinKM** [reply]
Did not a deep dive yet. But nzbparser seems a helper class and has no access to "cfg". So this means i have to add new parameters to "process_nzb_archive_file". Can imagine not acceptable.

I use already a custom docker image (inheritance of linuxserver/sabnzbd)
And patching while building also works
`patch /app/sabnzbd/sabnzbd/nzbparser.py < /scripts/nzbparser.py.patch && \`
But this means maintenance if the source code deviate to much.

**Reply by puzzledsab** [reply]
I think you can just import it. `import sabnzbd.cfg as cfg`. You shouldn't have any problems importing new changes when you have only changed 2 lines, though. That part of the code isn't likely to change much.

**Reply by EdwinKM** [reply]
> I think you can just import it. `import sabnzbd.cfg as cfg`. You shouldn't have any problems importing new changes when you have only changed 2 lines, though. That part of the code isn't likely to change much.

Thanks, the commit example was useful. Also needed to figure out how to create the PR -> https://github.com/sabnzbd/sabnzbd/pull/2485.
The validation seems not happy. Guess something to do tomorrow
